### PR TITLE
docs: reduce the amount of detail in the open/close port methods

### LINKED ISSUE #918: Improve open_port / close_port docs
The docstrings for [`open_port`](https://github.com/canonical/operator/blob/eb8e25a5314ca2063f4689fa829c8bdca652ce1f/ops/model.py#L605) and `close_port` have a bunch of Juju detail that probably shouldn't be here, and some of it isn't correct -- for example, the bit about "registering intent ... juju expose" is true for VM charms but not for K8s sidecar charms.

I think what we want to do here is improve our [Juju docs about open-port](https://juju.is/docs/sdk/hook-tool#heading--open-port) and then link to those, to avoid documenting too much Juju behaviour in `ops` docstrings.

### PR DESCRIPTION
**Pull request by tonyandrewmeyer**
Simplify the `open_port()` and `close_port()` documentation:

* Link to the Juju documentation ([soon to also be updated](https://chat.charmhub.io/charmhub/pl/heqdxrtrkjyotdn8iikcpegbxh)) to minimise the amount of syncing required
* Note that the behaviour differs between K8s sidecar and machine charms
* Remove the note that `juju expose`/`juju unexpose` is required (true for machine, not true for k8s sidecar)

Fixes #918

### CODE REVIEW

File: ops/model.py [hunk 1]
```diff
@@ -597,14 +597,11 @@ def open_port(self, protocol: typing.Literal['tcp', 'udp', 'icmp'],
...
 
-        Calling this registers intent with Juju that the application should be
-        accessed on the given port, but the port isn't actually opened
-        externally until the admin runs "juju expose".
-
-        On Kubernetes sidecar charms, the ports opened are not strictly
-        per-unit: Juju will open the union of ports from all units.
-        However, normally charms should make the same open_port() call from
-        every unit.
+        Some behaviour, such as whether the post is opened externally without
```
Comment (benhoyt) [inline ops/model.py:600]:
```suggestion
        Some behaviour, such as whether the port is opened externally without
```

File: ops/model.py [hunk 2]
```diff
@@ -619,10 +616,11 @@ def close_port(self, protocol: typing.Literal['tcp', 'udp', 'icmp'],
                    port: Optional[int] = None):
         """Close a port with the given protocol for this unit.
 
-        On Kubernetes sidecar charms, Juju will only close the port once the
-        last unit that opened that port has closed it. However, this is
-        usually not an issue; normally charms should make the same
-        close_port() call from every unit.
+        Some behaviour, such as whether the post is closed externally without
```
Comment (benhoyt) [inline ops/model.py:619]:
```suggestion
        Some behaviour, such as whether the port is closed externally without
```

**Review by benhoyt — CHANGES_REQUESTED** [review]
Changes look fine, just a typo, and the docs build seems to be failing (confirms my dislike of ReST).

**Comment by tonyandrewmeyer** [comment]
> the docs build seems to be failing (confirms my dislike of ReST).

Ah, my bad. I built the docs after doing `open_port()` to make sure that I had the link syntax correct, and then lazily did not do that after making the `close_port()` change, and it's angry that the external link is there twice. Will fix.

**Review by benhoyt — APPROVED** [review]

**Review by jnsgruk — APPROVED** [review]

**Pull request merged**